carta/cpp/CartaLib/IImage.h

```diff
@@ -319,6 +321,9 @@ class ImageInterface
     virtual const VI &
     dims() const = 0;
 
+    /// return the index of the specified axis in the image.
+    virtual int getAxisIndex( Carta::Lib::AxisInfo::KnownType axis ) const = 0;
+
```

Comment:
Does this have to be part of the interface class? I think the functionality could be entirely implemented already using just the public methods of this class.